app/src/main/java/fr/free/nrw/commons/auth/AuthenticatedActivity.java

```diff
@@ -24,87 +22,35 @@
-                return null;
-            } catch (AuthenticatorException e) {
-                e.printStackTrace();
-                return null;
-            } catch (IOException e) {
-                e.printStackTrace();
-                return null;
-            }
-        }
+    private void getAuthCookie(Account account, AccountManager accountManager) {
+        Single.fromCallable(() -> accountManager.blockingGetAuthToken(account, "", false))
+                .subscribeOn(Schedulers.io())
+                .doOnError(Timber::e)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        cookie -> onAuthCookieAcquired(cookie),
+                        throwable -> onAuthFailure());
     }
-    
-    private class AddAccountTask extends AsyncTask<Void, String, String> {
-        private AccountManager accountManager;
-        public AddAccountTask(AccountManager accountManager) {
-            this.accountManager = accountManager;
-        }
-        
-        @Override
-        protected void onPostExecute(String result) {
-            super.onPostExecute(result);
-            if(result != null) {
-                Account[] allAccounts =accountManager.getAccountsByType(accountType);
-                Account curAccount = allAccounts[0];
-                GetAuthCookieTask getCookieTask = new GetAuthCookieTask(curAccount, accountManager);
-                getCookieTask.execute();
-            } else {
-                onAuthFailure();
-            }
-        }
 
-        @Override
-        protected String doInBackground(Void... params) {
-            AccountManagerFuture<Bundle> resultFuture = accountManager.addAccount(accountType, null, null, null, AuthenticatedActivity.this, null, null);
-            Bundle result;
-            try {
-                result = resultFuture.getResult();
-            } catch (OperationCanceledException e) {
-                e.printStackTrace();
-                return null;
-            } catch (AuthenticatorException e) {
-                // TODO Auto-generated catch block
-                e.printStackTrace();
-                return null;
-            } catch (IOException e) {
-                // TODO Auto-generated catch block
-                e.printStackTrace();
-                return null;
-            }
-            if(result.containsKey(AccountManager.KEY_ACCOUNT_NAME)) {
-                return result.getString(AccountManager.KEY_ACCOUNT_NAME);
-            } else {
-                return null;
-            }
-            
-        }
+    private void addAccount(AccountManager accountManager) {
+        Single.just(accountManager.addAccount(accountType, null, null, null, AuthenticatedActivity.this, null, null))
+                .subscribeOn(Schedulers.io())
+                .map(AccountManagerFuture::getResult)
+                .doOnEvent((bundle, throwable) -> {
+                    if (bundle.containsKey(AccountManager.KEY_ACCOUNT_NAME)) {
```

Comment:
Doesnt this do the reverse of what we want?  If the bundle contains an account name it should be a success & not an exception, surely?